content/blogposts/2023/copilot-vs-cody.md

```diff
@@ -0,0 +1,229 @@
+
+## Scenario: Learning about a codebase
+
+For our first scenario, let's place ourselves in the shoes of a new developer onboarding onto a codebase. Without a code AI assistant, the developer would likely first look at the `readme.md` file to see if it gives a good overview or perhaps reach out to the maintainer of the codebase for a walkthrough.
+
+In my personal experience, the `readme.md` files of projects are often hit or miss. They may include an amazing overview and instructions, or they may be completely blank and everything in between. With a code AI assistant that supports chat functionality, we can ask it to tell us what a codebase does. Let's try it with Cody and Copilot to see which one does better.
+
+Our query will be: **"What does this application do?"**
+
+![Explain what a codebase does](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/explain-what-codebase-does.png)
+
+Both Cody and Copilot were able to give us an answer to our query. Cody is on the left-hand side, and Copilot is on the right, and we'll use this format throughout the post. Copilot told us about the different technologies used in the repository, while Cody told us the tech stack, APIs used, as well as the major features of the application.
+
+**Cody also cites its sources**, and we can see in the above screenshot that Cody used 14 files from our codebase to get enough context to generate a detailed response. Copilot, on the other hand, does not provide visibility into how it came up with the answer.
+
+I think we can give Cody the win in our first scenario. After round one, the score stands at:
+
+Cody - 1
+
+Copilot - 0
+
+## Scenario: Setting up the development environment
+
+Now that our developer knows a little bit about what the application does, the next logical thing for them to do is to figure out how to set up the local development environment so that they can run the application and make code changes themselves. Without a code AI assistant, the developer would again likely search for a `readme.md` file or similar instructions on how to get set up. But with Cody and Copilot ready, we can ask them how to get this application running locally.
+
+The query we'll pose is: **"How do I start this app up in development mode?"**
+
+![Setup development environment](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/setup-dev-environment.png)
+
+Both Cody and Copilot were able to give us a response to this question as well. The difference between the two responses again highlights the importance of having the right context.
+
+Cody read 12 files to generate the response and walked the developer through installing the required JavaScript and PHP dependencies, setting up an `.env` file, creating a local database, running a database migration to set up the correct tables and seed data, and finally starting the server.
```

Comment:
```suggestion
Cody read 12 files to generate the response and walked us through installing the required JavaScript and PHP dependencies, setting up an `.env` file, creating a local database, running a database migration to set up the correct tables and seed data, and finally starting the server.
```

Reply:
Just a general suggestion: I think it reads better if we put ourselves and the reader in the shoes of the developer and use "we"/"us". If we make this change, we should update the language throughout the post.